tests/jsunit/event_test.js

```diff
@@ -392,3 +392,154 @@ function test_varBackard_runForward() {
+  assertEquals(1, filteredEvents[1].newCoordinate.x);
+  assertEquals(1, filteredEvents[1].newCoordinate.y);
+  eventTest_tearDownWithMockBlocks();
+}
+
+function test_events_filterDifferentBlocks() {
+  eventTest_setUpWithMockBlocks();
+  var block1 = workspace.newBlock('field_variable_test_block', '1');
+  var block2 = workspace.newBlock('field_variable_test_block', '2');
+  var events = [
+    new Blockly.Events.BlockCreate(block1),
+    new Blockly.Events.BlockMove(block1),
+    new Blockly.Events.BlockCreate(block2),
+    new Blockly.Events.BlockMove(block2)
+  ];
+  var filteredEvents = Blockly.Events.filter(events, true);
+  assertEquals(4, filteredEvents.length);  // no event should have been removed.
+  eventTest_tearDownWithMockBlocks();
+}
+
+function test_events_mergeMove() {
+  eventTest_setUpWithMockBlocks();
+  var block1 = workspace.newBlock('field_variable_test_block', '1');
+  var events = [
+    new Blockly.Events.BlockMove(block1)
+  ];
+  block1.xy_.translate(1, 1);
+  events.push(new Blockly.Events.BlockMove(block1));
+  var filteredEvents = Blockly.Events.filter(events, true);
+  assertEquals(1, filteredEvents.length);
+  assertEquals(1, filteredEvents[0].oldCoordinate.x);
+  assertEquals(1, filteredEvents[0].oldCoordinate.y);
+  eventTest_tearDownWithMockBlocks();
+}
+
+function test_events_mergeChange() {
+  eventTest_setUpWithMockBlocks();
+  var block1 = workspace.newBlock('field_variable_test_block', '1');
+  var events = [
+    new Blockly.Events.Change(block1, 'field', 'VAR', 'item', 'item1'),
+    new Blockly.Events.Change(block1, 'field', 'VAR', 'item1', 'item2')
+  ];
+  var filteredEvents = Blockly.Events.filter(events, true);
+  assertEquals(1, filteredEvents.length);
+  assertEquals('item', filteredEvents[0].oldValue);
+  assertEquals('item2', filteredEvents[0].newValue);
+  eventTest_tearDownWithMockBlocks();
+}
+
+function test_events_mergeUi() {
+  eventTest_setUpWithMockBlocks();
+  var block1 = workspace.newBlock('field_variable_test_block', '1');
+  var block2 = workspace.newBlock('field_variable_test_block', '2');
+  var block3 = workspace.newBlock('field_variable_test_block', '3');
+  var events = [
+    new Blockly.Events.Ui(block1, 'commentOpen', 'false', 'true'),
+    new Blockly.Events.Ui(block1, 'click', 'false', 'true'),
+    new Blockly.Events.Ui(block2, 'mutatorOpen', 'false', 'true'),
+    new Blockly.Events.Ui(block2, 'click', 'false', 'true'),
+    new Blockly.Events.Ui(block3, 'warningOpen', 'false', 'true'),
+    new Blockly.Events.Ui(block3, 'click', 'false', 'true')
+  ];
+  var filteredEvents = Blockly.Events.filter(events, true);
+  assertEquals(3, filteredEvents.length);
+  assertEquals('commentOpen', filteredEvents[0].element);
+  assertEquals('mutatorOpen', filteredEvents[1].element);
+  assertEquals('warningOpen', filteredEvents[2].element);
+  eventTest_tearDownWithMockBlocks();
+}
+
+/**
+ * Tests that the creation/movement of many unique blocks does not have severe
```

Comment:
Timing is currently something we keep track of by using "airstrike" and "spaghetti" in the playground.  My concern with this test is that there's no way for it to fail, so it adds no information to test output but will slow down our presubmit process.  

For that reason I'd like to remove it from this test file, and I'll keep an eye on performance through the playground instead.